NuRadioMC/examples/simulate_effective_ice_temperature/ice_integrator.py

```diff
@@ -0,0 +1,258 @@
+    sim.add(obs2)
+
+    obs3 = radiopropa.Observer()
+    obsz3 = radiopropa.ObserverSurface(radiopropa.Plane(radiopropa.Vector3d(0, 0, 10), radiopropa.Vector3d(0, 0, 1)))
+    obs3.add(obsz3)
+    obs3.setDeactivateOnDetection(True)
+    sim.add(obs3)
+
+    if in_file:
+        # Output
+        if os.path.isfile(filename):
+            os.remove(filename)
+
+        output = radiopropa.HDF5Output(filename, radiopropa.Output.Trajectory3D)
+        output.setLengthScale(radiopropa.meter)
+        #output.enable(radiopropa.Output.CurrentAmplitudeColumn)
+        output.enable(radiopropa.Output.SerialNumberColumn)
+        sim.add(output)
+
+    # Source - Antenna at z_antenna at which the ray tracing is started
+    source = radiopropa.Source()
+    source.add(radiopropa.SourcePosition(radiopropa.Vector3d(0, 0, z_antenna)))
+    source.add(radiopropa.SourceAmplitude(1))
+    source.add(radiopropa.SourceFrequency(freq))
+    z = np.cos(theta / units.deg * radiopropa.deg)
+    x = np.sin(theta / units.deg * radiopropa.deg)
+    source.add(radiopropa.SourceDirection(radiopropa.Vector3d(x, 0 , z)))
+    sim.setShowProgress(False)
+
+    if in_file:
+        sim.run(source, 1)
+        return filename
+    else:
+        ray = source.getCandidate()
+        sim.run(ray, True)
+        # Do something with the ray ...
+
+
+def get_trace(theta, z_antenna, freq=403 * 1e6):
+    """ Wrapper around get_trace_file """
+    file = get_trace_file(theta, z_antenna, in_file=True, freq=freq)
+    return get_angles_rhos(file)
+
+
+def temperature_integral(zenith, z_antenna, freq=400 * units.MHz, model="GL3"):
+    """ Return the integral of the effective temperature along the ray path.
+
+    This integral takes into accout the attenuation of the signal along the ray path.
+    The effective temperature is calculated as the sum of the temperature at each
+    point along the ray path weighted by the attenuation factor at that point (emissivity).
+
+    Parameters
+    ----------
+    zenith : float
+        Zenith angle of the ray
+    z_antenna : float
+        Depth of the antenna in meters
+    freq : float
+        Frequency of the signal
+    model : str
+        Ice attenuation model to use
+
+    Returns
+    -------
+    eff_temperature : float
+        Effective temperature at the antenna
+    """
+    # get the track for respective zenith direction
+    # (distance vs depth)
+    _, radius, depth_raytracing, distance_raytracing, reflected_power, _ = get_trace(zenith, z_antenna)
+
+    # get the temperature at each of the track points
+    depth_interp = interp1d(distance_raytracing, depth_raytracing, fill_value="extrapolate")
+    power_interp = interp1d(distance_raytracing, reflected_power, fill_value="extrapolate")
+
+    distance = np.linspace(0, 30000, 300000) * units.m
+    d_distance = distance[1] - distance[0]
+
+    depth = depth_interp(distance)
+    ref = power_interp(distance)
+    # manually set cap stuff
+    ref[depth < z_min] = 0
```

Comment:
did you test if that has no influence on the obtained temperature at the antenna? (i.e. - quickest solution - by commenting this line and setting the bottom to something like -6k meters ... ((or anything >>3km ... that way you shouldn't have to worry about the shorter attenuation length in rock...)))

Reply:
Okay I removed the capping and plotted the `attenuation_factor` for downwards going rays. As you can see attenuation factor for the most vertical rays at the rock bottom is < 0.1 (i.e., a factor of 10+ attenuated. But it is not equal 0.
![image](https://github.com/user-attachments/assets/85e20b93-86f9-4a48-b715-eccebceb03c6)

Reply:
Linear color map but everything below 0.01 is cut out.
![image](https://github.com/user-attachments/assets/89801ad0-7ecc-40c5-9213-07a546d6b780)

Reply:
Here is the limit 0.03. So < 3% comes from the rock bottom (for the given attenuation model).
![image](https://github.com/user-attachments/assets/f0fec1e5-cc3b-4637-880a-f2d172c9aa7f)

Reply:
So what is our conclusion. Should we just pretend the ice is infinitive deep and (linearly?) extrapolating the temperature/attenuation profiles since the contributions do not matter anyhow?

Reply:
Also my statement is incorrect. Not less than 3% come from the bedrock but 3% from the bedrock arrive at the antenna.

Reply:
Okay, when I plot what I believe is a good indicate for how much the temp from a certain volume element contributes, than this is very similar to the plot above (again cut at 0.03). The color axis shows this:
```
        eff_temp = reflection_coef * att_factor * temp_env / l_att * d_distance

        eff_temp = 1 - np.cumsum(eff_temp) / np.sum(eff_temp)

        eff_temp[eff_temp < 0.03] = np.nan
```
I am actually surprised that we still have 3 percent contribution from the bedrock. I believe this is actually wrong. Hmmm. So probably we have to change things there.
![image](https://github.com/user-attachments/assets/101d2882-a155-411b-9081-408d128ffa1a)

Reply:
Hi Felix! Thanks for this very comprehensive follow-up work. I just don't understand: would you expect much less than 3%? Back of the envelope we should be something between 3-4 absorption lengths, i.e. I do expect e**-3...4 = 2 to 5 % of the temperature should come from there, no?

Reply:
> So what is our conclusion. Should we just pretend the ice is infinitive deep and (linearly?) extrapolating the temperature/attenuation profiles since the contributions do not matter anyhow?

I would guess that the attenuation length of rock for radio is very short. I would therefore keep the bottom temperature fixed at the bottom instead of extrapolating. You can continue for a short bit with high attenuation (=high emissivity) or just leave the attenuation constant and continue more (that's not what is physically happening but it should give you the same result if I am not mistaken)

Reply:
I was playing with something like this around yesterday evening but did not understand the results. Will try to revisit it now. So I am happy with receiving 2-5% of the radiation from the bedrock, but I thought this 2-5% should mean nothing for the eff. Temperature at the antenna.

Reply:
Ok, well... 2-5% means 5 to 13 degrees more in temperature, or am I mistakenb? ... so it is not huge, but nonzero